frontend/ui/impl/src/test/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitterTest.java

```diff
@@ -0,0 +1,123 @@
+  private static String errorReportWithStackTraceAndTitle;
+  private static String systemLineSeparator = "\n";
+
+  @BeforeClass
+  @SneakyThrows
+  public static void setUpStatic() {
+    defaultErrorReport = IOUtils.resourceToString("/error_report_uri.txt", StandardCharsets.UTF_8).strip();
+    defaultErrorReportWithTemplate = IOUtils.resourceToString("/error_report_uri_with_template.txt",
+        StandardCharsets.UTF_8).strip();
+    errorReportWithStackTraceAndTitle = IOUtils.resourceToString("/error_report_uri_with_stacktrace_title.txt",
+        StandardCharsets.UTF_8).strip();
+  }
+
+  @Before
+  public void setUp() {
+    Whitebox.setInternalState(GitMacheteErrorReportSubmitter.class, errorReportSubmitterLogger);
+    Whitebox.setInternalState(SystemUtils.class, "OS_NAME", "Mock OS X");
+    Whitebox.setInternalState(SystemUtils.class, "OS_VERSION", "Hehe");
+
+    PowerMockito.stub(PowerMockito.method(System.class, "lineSeparator"))
+            .toReturn(systemLineSeparator);
+
+    PowerMockito.mockStatic(ApplicationInfo.class);
+    PowerMockito.mockStatic(PluginId.class);
+
+    val applicationInfo = PowerMockito.mock(ApplicationInfo.class);
+    PowerMockito.when(applicationInfo.getFullApplicationName()).thenReturn("mocked IntelliJ idea");
+    PowerMockito.stub(PowerMockito.method(ApplicationInfo.class, "getInstance"))
+        .toReturn(applicationInfo);
+  }
+
+  private IdeaLoggingEvent[] getMockEvents(String throwableText) {
+    val customThrowable = new Throwable() {
+      @Override
+      public void printStackTrace(PrintWriter s) {
+        s.print(throwableText);
+      }
+    };
+
+    return new IdeaLoggingEvent[]{
+        new IdeaLoggingEvent(/* message */ "", customThrowable)
+    };
+  }
+
+  private String invokeGetReportBody(GitMacheteErrorReportSubmitter errorReportSubmitter, IdeaLoggingEvent[] events)
+      throws Exception {
+    String reportBody = Whitebox.invokeMethod(errorReportSubmitter, "getReportBody", events, /* additionalInfo */ null);
+
+    Mockito
+        .verify(errorReportSubmitterLogger,
+            never().description("Reading the resource file with the bug template should NOT throw exceptions"))
+        .error(anyString(), (Throwable) any());
+
+    return reportBody;
+  }
+
+  @Test
+  public void shouldConstructUriWithTemplate() throws Exception {
+    val events = getMockEvents(/* throwableText */ "");
+    val reportBody = invokeGetReportBody(reportSubmitter, events);
+
+    URI uriWithReport = Whitebox.invokeMethod(reportSubmitter, "constructNewGitHubIssueUri", events, reportBody);
+    Assert.assertEquals("URI should match default with report body", defaultErrorReportWithTemplate, uriWithReport.toString());
+  }
+
+  @Test
+  public void shouldConstructUriWithNoTemplate() throws Exception {
+    val events = getMockEvents(/* throwableText */ "");
+
+    URI uriNoReport = Whitebox.invokeMethod(reportSubmitter, "constructNewGitHubIssueUri", events, /* report Body */ null);
```

Comment:
Another simplification: once you make `constructNewGitHubIssueUri` package-private, it seems that the test still passes with:
```suggestion
    URI uriNoReport = reportSubmitter.constructNewGitHubIssueUri(events, /* report Body */ null);
```

Reply:
Sure it will pass, since Whitebox just lets us invoke the private method. I'd leave it as it is and keep the encapsulation.